### PR TITLE
ci: pre-commit hook to check the commit matches with the convention

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

1. Git pre-commit hook for the [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/) conformity check:
https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional

2. Github action for automatic generation of `CHANGELOG.md`:
https://github.com/googleapis/release-please

> Warning! I haven't tried `release-please` yet. Will test it after merging this PR

image

Screenshot 2023-04-24 at 14 12 30

### CODE REVIEW

**Review by tukantje — APPROVED** [review]
Looks awesome - not sure how to test it out but I can't see any issues personally.

**Review by nenadV91 — APPROVED** [review]
Looks good. I'm just wondering if it creates a new release PR for every merge to `develop` or it can update one release PR, which is what we want I think.

**Review by anxolin — COMMENTED** [review]
👍 Cool

File: commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  extends: ['@commitlint/config-conventional'],
```
Comment (alfetopito) [inline commitlint.config.js:2]: Does it mean that every commit will have to follow this standard?
That will be annoying for individual PRs
Can we have this only for PRs and/or merges to develop & main?

**Pull request merged**